### PR TITLE
Return human-readable description by ComparisonRule.toString()

### LINKED ISSUE #373: Report comparison rules in plain English
**Is your feature request related to a problem? Please describe.**
Current comparison rules reported as EQUAL_TO. To increase readability we should convert them to plain text, **equal to**

### PR DESCRIPTION
**Pull request by ikalinin1**
Closes #373

### CODE REVIEW

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/ComparisonRule.java [hunk 1]
```diff
@@ -94,6 +96,12 @@ public static ComparisonRule parse(String sign)
                 .findFirst();
     }
 
+    @Override
+    public String toString()
+    {
+        return asString;
```
Comment (valfirst) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/ComparisonRule.java]:
```java
return name().toLowerCase().replace('_', ' ');
```
?

File: vividus-plugin-html/src/main/java/org/vividus/bdd/steps/html/HtmlSteps.java [hunk 2]
```diff
@@ -56,6 +48,36 @@
...
+     * @see <a href="https://jsoup.org/apidocs/org/jsoup/select/Selector.html"><i>Jsoup Selector API</i></a>
+     * @return true if elements quantity corresponds to rule, false otherwise
+     */
+    @Then("number of elements found by CSS selector `$cssSelector` in HTML `$html` is $comparisonRule `$quantity`")
+    public boolean doesElementByCssSelectorExist(String cssSelector, String html, ComparisonRule comparisonRule,
+            int number)
+    {
+        Elements elements = findElements(html, cssSelector);
+        return softAssert.assertThat(String.format("Number of elements found by CSS selector '%s' is %s %d",
+                cssSelector, comparisonRule, number), elements.size(), comparisonRule.getComparisonRule(number));
```
Comment (valfirst) [inline vividus-plugin-html/src/main/java/org/vividus/bdd/steps/html/HtmlSteps.java]: do you really need to add `comparisonRule` and `number` to assertion description? it could be excessive since similar piece of information is added by Matcher

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/ComparisonRule.java (hunk 1 above)
Comment (ikalinin1) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/ComparisonRule.java]: I considered an extra code execution for each toString() method call, when values already defined, as an overhead

File: vividus-plugin-html/src/main/java/org/vividus/bdd/steps/html/HtmlSteps.java (hunk 2 above)
Comment (ikalinin1) [reply vividus-plugin-html/src/main/java/org/vividus/bdd/steps/html/HtmlSteps.java]: I think we should pick one solution. BaseValidation uses this approach.
If we will rely on matcher, the PR is obsolete, but we should refactor basevalidations' code

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/ComparisonRule.java [hunk 3]
```diff
@@ -94,6 +94,11 @@ public static ComparisonRule parse(String sign)
                 .findFirst();
     }
 
+    public String toString()
+    {
+        return super.toString().replaceAll("_", " ").toLowerCase();
```
Comment (valfirst) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/ComparisonRule.java]: since JDK 11 is still our baseline, `.replaceAll("_", " ")` -> `.replace('_', ' ')`
